src/Components/ArtworkGrid/ArtworkGridContext.tsx

```diff
@@ -13,20 +18,41 @@ interface ArtworkGridContextProps {
   hideLotLabel?: boolean
 
   saveOnlyToDefaultList?: boolean
+
+  collectorSignals: ReturnType<typeof useCollectorSignals>
 }
 
 const ArtworkGridContext = createContext<ArtworkGridContextProps>({
   isAuctionArtwork: false,
   hideLotLabel: false,
   saveOnlyToDefaultList: false,
+  collectorSignals: {},
 })
 
-export const ArtworkGridContextProvider: React.FC<ArtworkGridContextProps> = ({
+type ArtworkGridContextProviderProps = {
+  artworksConnection?: ArtworkGridContext_artworksConnection$key | null
+  me?: ArtworkGridContext_me$key | null
+} & Omit<ArtworkGridContextProps, "collectorSignals">
+
+export const ArtworkGridContextProvider: React.FC<ArtworkGridContextProviderProps> = ({
+  artworksConnection,
+  me,
   children,
   ...rest
 }) => {
+  const artworksData = useFragment(
+    ARTWORKS_CONNECTION_FRAGMENT,
+    artworksConnection
+  )
+  const meData = useFragment(ME_FRAGMENT, me)
+
+  const collectorSignals = useCollectorSignals({
+    artworks: extractNodes(artworksData),
```

Comment:
I will comment here because it'll be easier to see, we pass artwork data here, so I truly believe these data elements should be extracted inside `useCollectorSignals` with `useFragment`.

If you do the same that you do in the hook level will be better IMO, especially for this component tidiness, something like:

```jsx
export const ArtworkGridContextProvider: React.FC<ArtworkGridContextProviderProps> = ({
  artworksConnection,
  me,
  children,
  ...rest
}) => {
  const collectorSignals = useCollectorSignals({
    artworksConnection,
    me,
  })
  ...
```

in the hook:

```jsx
export function useCollectorSignals({
  artwork,
  artworks,
  me,
}: UseCollectorSignalsSingleArtwork | UseCollectorSignalsMultipleArtworks):
  | SignalResultForArtworks
  | SignalResult {
  const signalsPartnerOffersEnabled = useFeatureFlag(
    "emerald_signals-partner-offers"
  )

  const artworkData = useFragment()
  const artworksConnectionData = useFragment()
  const meData = useFragment()
  
  // all the code below will need to be adapted but the relay data should come from the hook itself, not drilled
}

```

Reply:
With @erikdstock having gone this original direction, and your thinking here i'm swayed away from neutrality. 

@erikdstock - thoughts?

Reply:
The pass went quick for implementation, but tests took forever. Testing with renderHook ran into problems because i needed to generate a mock payload with fragments, but mock payload couldn't be handled by `useFragment()`. My theory is that the mock payload expects to be resolved by a `mockEnv.mock.resolveMostRecentOperation()` call which is probably where the fragment bits get handled. I got around this by mocking `useFragment()` to just return the data after several other abortive attempts like `@relay(mask: true)`.

Overall I'm fine with this or the previous approach. LMK if i should merge or revert.
https://github.com/artsy/force/pull/14132/commits/d8d16abaf68c01257eead2da981f5512274d6e70